packages/@aws-cdk/cloudformation-diff/test/diff-template.test.ts

```diff
@@ -624,6 +624,19 @@ test('metadata changes are rendered in the diff', () => {
   expect(differences.resources.differenceCount).toBe(1);
 });
 
+test('diff complains if it receives an undefined template', async () => {
+  // GIVEN
+  const currentTemplate: any = undefined;
+
+  // WHEN
+  const newTemplate = {
+    Resources: {},
+  };
+
+  // THEN
+  expect(() => fullDiff(currentTemplate, newTemplate)).toThrow(/trying to diff a template that is not defined/);
```

Comment:
From this code I can get why there is this check. But how would an actual user end up with this? How can they fix it?

Reply:
A user would never end up with this. I added it because diff is design to work on arbitrary json objects, but it achieves this by typing things as `any`, which allows `undefined`. `undefined` breaks it though, so I figured we could be a bit more defensive with that. It's not doing anything useful though, it's just moving the crash earlier, so I can remove this

Reply:
The reason this change got through is it looked innocuous and it didn't break any tests, because we had just never tested fixed-template diffs before